rasa/core/story_conflict.py

```diff
@@ -0,0 +1,252 @@
+from typing import List, Optional, Dict, Text
+
+from rasa.core.actions.action import ACTION_LISTEN_NAME
+from rasa.core.domain import PREV_PREFIX, Domain
+from rasa.core.events import ActionExecuted, Event
+from rasa.core.featurizers import MaxHistoryTrackerFeaturizer
+from rasa.nlu.constants import MESSAGE_INTENT_ATTRIBUTE
+from rasa.core.training.generator import TrackerWithCachedStates
+
+
+class StoryConflict:
```

Comment:
I think there is no need to make all these methods static, they can just live as functions in this module

Reply:
I don't agree because these functions all seem todo things which are belong the responsibilities of `StoryConflict`. I also don't think that all of these functions are truly static. E.g. why is `domain` and `max_history` passed as parameters and not referenced via `self.<attribute>`?